### PR TITLE
feat(appeals): add lpa reference change page and api

### PR DESCRIPTION
**Pull request by AlexMoulin**
## Issue ticket number and link
https://pins-ds.atlassian.net/browse/BOAT-1111

## Type of change 🧩

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [x] I have made corresponding changes to the documentation
- [x] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

### CODE REVIEW

**Review by fabfab-kc — APPROVED** [review]
LGTM!

**Review by fabfab-kc — APPROVED** [review]

File: appeals/web/src/server/appeals/appeal-details/__tests__/appeal-details.test.js [hunk 1]
```diff
@@ -457,6 +457,32 @@ describe('appeal-details', () => {
...
+
+				const caseDetailsResponse = await request.get(`${baseUrl}/1`);
+				try {
+					const notificationBannerElementHTML = parseHtml(caseDetailsResponse.text, {
+						rootElement: notificationBannerElement
+					}).innerHTML;
+					expect(notificationBannerElementHTML).toMatchSnapshot();
+					expect(notificationBannerElementHTML).toContain('LPA application reference updated');
+				} catch (error) {
+					console.log('There are no notification banner elements in the html', error);
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/__tests__/appeal-details.test.js:483]: Very minor, you may want to remove this or replace with a `logger` call

File: appeals/web/src/server/appeals/appeal-details/appellant-case/appellant-case.router.js [hunk 2]
```diff
@@ -30,6 +31,9 @@ router.use('/valid', outcomeValidRouter);
 router.use('/invalid', outcomeInvalidRouter);
 router.use('/incomplete', outcomeIncompleteRouter);
 
+//Change pages
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/appellant-case/appellant-case.router.js:34]: This comment seems unnecessary

File: appeals/web/src/server/appeals/appeal-details/change-lpa-reference/change-lpa-reference.controller.js [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+		errors,
+		params: { appealId }
+	} = request;
+	const currentUrl = request.originalUrl;
+
+	const origin = currentUrl.split('/').slice(0, -2).join('/');
+
+	console.log(origin);
+
+	//const linkSuffix =  ? `$`
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/change-lpa-reference/change-lpa-reference.controller.js:30]: This looks like it should be removed

File: appeals/web/src/server/appeals/appeal-details/change-lpa-reference/change-lpa-reference.controller.js [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+const renderChangeLpaReference = async (request, response) => {
+	const {
+		errors,
+		params: { appealId }
+	} = request;
+	const currentUrl = request.originalUrl;
+
+	const origin = currentUrl.split('/').slice(0, -2).join('/');
+
+	console.log(origin);
```
Comment (james-hill-kc) [inline appeals/web/src/server/appeals/appeal-details/change-lpa-reference/change-lpa-reference.controller.js:28]: This should probably be removed

**Review by james-hill-kc — COMMENTED** [review]
Looks good, just a few minor comments from me

**Pull request merged**